src/dolphin/atmosphere/ionosphere.py

```diff
@@ -0,0 +1,383 @@
+        path to the SLC file
+    tec_file: Filename
+        path to the tec file corresponding to slc date
+    lat: float
+        latitude of scene center
+    lon: float
+        longitude of scene center
+
+    Returns
+    -------
+    float
+        zenith TEC of the scene center in TECU.
+    """
+    time = oput.get_zero_doppler_time(slc_file)
+    utc_seconds = time.hour * 3600.0 + time.minute * 60.0 + time.second
+
+    vtec = get_ionex_value(tec_file=tec_file, utc_sec=utc_seconds, lat=lat, lon=lon)
+
+    return vtec
+
+
+def vtec2range_delay(
+    vtec: float, inc_angle: np.ndarray, freq: float, obs_type: str = "phase"
+):
+    """Calculate/predict the range delay in SAR from TEC in zenith direction.
+
+    Equation (6-11) from Yunjun et al. (2022).
+
+    Parameters
+    ----------
+    vtec: float
+        zenith TEC in TECU
+    inc_angle: np.ndarray
+        incidence angle at the ionospheric shell in deg
+    freq: float
+        radar carrier frequency in Hz.
+    obs_type: str
+        given the same iono, the impact on offset (amplitude) and phase is reversed.
+
+    Returns
+    -------
+    np.ndarray
+        predicted range delay in meters
+    """
+    # ignore no-data value in inc_angle
+    if isinstance(inc_angle, np.ndarray):
+        inc_angle[inc_angle == 0] = np.nan
+
+    # convert to TEC in LOS based on equation (3) in Chen and Zebker (2012)
+
+    # Equation (26) in Bohm & Schuh (2013) Chapter 2.
+    n_iono_group = 1 + K * vtec * 1e16 / freq**2
+
+    # Equation (8) in Yunjun et al. (2022, TGRS)
+    ref_angle = (
+        np.arcsin(1 / n_iono_group * np.sin(inc_angle * np.pi / 180)) * 180 / np.pi
+    )
+
+    tec = vtec / np.cos(ref_angle * np.pi / 180.0)
+
+    # calculate range delay based on equation (1) in Chen and Zebker (2012)
+    range_delay = (tec * 1e16 * K / (freq**2)).astype(np.float32)
+
+    # group delay = phase advance * -1
+    if obs_type != "phase":
+        range_delay *= -1.0
+
+    return range_delay
+
+
+def get_ionex_value(
+    tec_file: Filename, utc_sec: float, lat: float, lon: float
+) -> float:
+    """Get the TEC value from input IONEX file for the input lat/lon/datetime.
+
+    Parameters
+    ----------
+    tec_file: Filename
+        path of local TEC file
+    utc_sec: float
+        UTC time of the day in seconds
+    lat: float
+        latitude in degrees
+    lon: float
+        longitude in degrees
+
+
+    Returns
+    -------
+    float
+        vertical TEC value in TECU
+    """
+    # time info
+    utc_min = utc_sec / 60.0
+
+    # read TEC file
+    mins, lats, lons, tec_maps = read_ionex(tec_file)
+
+    # interpolate between consecutive rotated TEC maps
+    # reference: equation (3) in Schaer et al. (1998)
+
+    # prepare interpolation functions in advance to speed up
+    ind0 = np.where((mins - utc_min) <= 0)[0][-1]
+    ind1 = ind0 + 1
+
+    interpfs = []
+    for i in [ind0, ind1]:
+        interpfs.append(
+            interpolate.interp2d(
```

Comment:
i think we'll want to change this due to the deprecation warning: https://github.com/opera-adt/COMPASS/issues/98